src/@types/topic.ts

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { EitherAnd } from "matrix-events-sdk";
+import { EitherAnd, Optional } from "matrix-events-sdk";
```

Comment:
Can we please not use more from the matrix-events-sdk given its borderline abandoned

Reply:
Should it gain a `@deprecated` flag?

Reply:
> I believe formally the spec core team owns it.

That would be very much news to me. I think you're right that it's abandoned, anyway.

(That said, it seems to me that it would make sense, as a maintenance strategy for the js-sdk, to factor out a lighter-weight library which just exposes some typescript types and maybe thin wrappers around the REST APIs: currently MatrixClient tries to be both high-level and low-level and it's confusing to use and maintain.)

Reply:
> (That said, it seems to me that it would make sense, as a maintenance strategy for the js-sdk, to factor out a lighter-weight library which just exposes some typescript types and maybe thin wrappers around the REST APIs

Agreed, though I'd want both packages shipping out of the same repo as needing to create 3 PRs to add an endpoint & logic & consumer would be a PITA.